### ISSUE TITLE
Can H.NotifyIcon be used in Winform

### ISSUE THREAD

**Issue by FreeVB** [HavenDV/H.NotifyIcon#96]
Can H.NotifyIcon be used in Winform?

**Comment by HavenDV** [comment]
Yes, sure. There is no separate implementation under WindowsForms, but there is a general implementation suitable for any platform. Here's an example code:
https://github.com/HavenDV/H.NotifyIcon/blob/master/src/apps/H.NotifyIcon.Apps.Console/Program.cs

**Comment by FreeVB** [comment]
I tested this method but encountered an error as indicated in the image below.
[WinFormsApp1.zip](https://github.com/HavenDV/H.NotifyIcon/files/11661339/WinFormsApp1.zip)
![error](https://github.com/HavenDV/H.NotifyIcon/assets/30093956/706a25ef-9688-4400-9382-c96b4a24d85c)

**Comment by FreeVB** [comment]
It seems that the reason why the ICON attribute must be set is that if it is empty, an error will occur.



Is it due to thread settings? Although I started normally, the form did not display. If I manually add this. Show(); Forms can be displayed, but they are not operable.

**Comment by HavenDV** [comment]
Yes, there may be issues here using TrayIconWithContextMenu. This class assumes that the process does not have a window. Please take a look at this code and try using regular TrayIcon
https://github.com/HavenDV/H.NotifyIcon/blob/master/src/libs/H.NotifyIcon/PopupMenus/TrayIconWithContextMenu.cs

**Comment by HavenDV** [comment]
Closing this as answered. WinForms can use the core TrayIcon API directly; TrayIconWithContextMenu is intended for windowless/native-menu scenarios and is not the best fit for a regular WinForms window. Use TrayIcon from the console sample pattern and set a valid icon before creating it.
